backend/node/views.py

```diff
@@ -0,0 +1,113 @@
+from django.shortcuts import render
+from rest_framework.generics import GenericAPIView
+from .models import Node
+# Create your views here.
+from rest_framework.decorators import api_view, permission_classes
+
+from rest_framework import response, status
+from rest_framework.decorators import api_view
+from rest_framework.permissions import AllowAny, IsAuthenticated, IsAuthenticatedOrReadOnly
+#import 
+import requests
+from rest_framework import authentication
+
+import base64
+
+"""
+I thought we didnt need generic view here
+
+"""
+
+credentialForConnect = {"username" : "hello", "password" : "world"}  # credential to connect
+credentialForDelete = {"", "", "", ""}
+
+
+# TODO:   this permission class DOES NOT WORK. get request still goes to my customAuthentication class in nodeFunction.py
+@api_view(['GET', 'DELETE'])
+@permission_classes((IsAuthenticatedOrReadOnly, ))
+def AcceptConnectionFromRemote(request, hostName):
+    """
+    https://stackoverflow.com/questions/38016684/accessing-username-and-password-in-django-request-header-returns-none
+
+    usage [remote]   GET    /service/node/remoteHostName
+          [local]    DELETE /service/node/remotehostname
+
+
+    NOTE, REFRACTOR IDEA, 
+          -we make remote host send 'Origin' header so we can shorten the path to    /service/node/
+          -having Origin is better for CustomAuthenticationCheck too because now I can query the host in db by their hostname
+    """
+
+    # TODO: need a way to restrict it only for LOCAL. its hard because this API doesnt require LOG IN
+    # NOTE, REQUIRE BASIC AUTH with username and password of this hostName for this request to be authenticated
+    if (request.method == "DELETE"):
+        # only our host can use this method.
+        # NOTE, TO DO THAT, there should be a way to check if this this server's heroku host == our heroku host
+
+        hostToBeDeleted = Node.objects.filter(hostName=hostName)
+
+        if hostToBeDeleted.exists():
+            Node.objects.filter(hostName=hostName).delete()
+            return response.Response(f"successfully severed the connection  ", status=status.HTTP_200_OK)
+        return response.Response(f"This host don't exist ", status=status.HTTP_404_NOT_FOUND)
```

Comment:
Like I mentioned before, we probably don't need a DELETE method since the server admin can just manage nodes in the admin panel, once you register in the node model.
```suggestion
```

Reply:
done